.github/workflows/triage-helper.yml

```diff
@@ -0,0 +1,27 @@
+name: Issue Triage Helper
+on:
+  issues:
+    types: [labeled]
+jobs:
+  routed:
+    runs-on: ubuntu-latest
+    if: >-
+      contains(
+        fromJSON('[
+          "Status: In Progress",
+          "Status: Backlog",
+          "Status: Needs More Information"
+        ]'),
+        github.event.label.name
+      )
+    steps:
+      - name: "Remove all 'Status: *' labels"
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_REPO: ${{ github.repository }}
+        run: |
+          labels_to_remove=$(gh issue view ${{ github.event.issue.number }} --json labels --jq '[.labels[].name | select(startswith("Status: ") and . != "${{ github.event.label.name }}")] | join(",")' )
```

Comment:
Per slack can we set `labels_to_remove` to all available `Status: *` labels defined in the repo, not just the ones currently applied? That seems the best way to fully avoid the race condition here.

Reply:
I don't think this would avoid the race condition at all. The job will run for any valid tasks everytime so there isn't a scenario you would end up with stray labels.

Reply:
Imagine this scenario:

1. Issue exists with label A.
2. Apply label B, action run (1) started, sees A.
3. Apply label C, action run (2) started, sees A.
4. Run (1) removes A, adds B.
5. Run (2) removes A, adds C.
6. Issue exists with with labels B and C.

Stray label race condition, no?

Reply:
Suggestion is:

1. Issue exists with label A.
1. Apply label B, action run (1) started.
1. Apply label C, action run (2) started.
1. Run (1) removes A C, adds B.
1. Run (2) removes A B, adds C.
1. Issue exists with with labels C.

No race, no stray. Last call to `gh issue edit` wins.